src/filetreediff.js

```diff
@@ -13,103 +15,136 @@ export class FileTreeDiffElement extends LitElement {
-    });
-
-    const generateDiffList = (diffArray, label) => {
-      return diffArray.length
-        ? html`
-            <span>${label}</span>
-            <ul>
-              ${repeat(
-                diffArray,
-                (f) => f.filename,
-                (f, index) =>
-                  html`<li>
-                    <a href=${f.urls.current}>${f.filename}</a>
-                    (<a href="${f.urls.current}?${DOCDIFF_URL_PARAM}=true"
-                      >diff</a
-                    >)
-                  </li>`,
-              )}
-            </ul>
-          `
-        : nothing;
-    };
+  handleToggleDiff(event) {
+    if (event.target.checked) {
+      document.dispatchEvent(
+        new CustomEvent(EVENT_READTHEDOCS_DOCDIFF_ADDED_REMOVED_SHOW),
+      );
+    } else {
+      document.dispatchEvent(new CustomEvent(EVENT_READTHEDOCS_DOCDIFF_HIDE));
+    }
+  }
+
+  getCurrentPageUrl() {
+    // Remove any query parameters to match against file URLs
+    const currentPath = window.location.pathname;
+    const currentOrigin = window.location.origin;
+    return `${currentOrigin}${currentPath}`;
+  }
 
+  render() {
     const diffData = objectPath.get(this.config, "addons.filetreediff.diff");
     if (!diffData) {
       return nothing;
     }
-    const diffAddedUrls = generateDiffList(diffData.added, "Added");
-    const diffDeletedUrls = generateDiffList(diffData.deleted, "Deleted");
-    const diffModifiedUrls = generateDiffList(diffData.modified, "Modified");
+
+    const currentUrl = this.getCurrentPageUrl();
+    const renderSection = (files, label) => {
+      if (!files.length) return nothing;
+      const emoji = label === "Added" ? "+ " : "± ";
+      return html`
+        <optgroup label="${label}">
+          ${files.map(
+            (f) => html`
+              <option
+                value=${f.urls.current}
+                ?selected=${f.urls.current === currentUrl}
+              >
+                ${emoji}${f.filename}
+              </option>
+            `,
+          )}
+        </optgroup>
+      `;
+    };
+
+    const hasCurrentFile = [...diffData.added, ...diffData.modified].some(
+      (f) => f.urls.current === currentUrl,
+    );
 
     return html`
-      <div>
-        ${iconFile.node[0]}
-        <div class="title">
-          Files changed in this version ${this.renderCloseButton()}
-        </div>
-        <div class="content">
-          ${diffAddedUrls} ${diffModifiedUrls} ${diffDeletedUrls}
+      <div class="file-dropdown">
+        <div class="diff-controls">
+          <label class="diff-checkbox">
+            <input
+              type="checkbox"
+              .checked=${this.docDiffEnabled}
+              @change=${this.handleToggleDiff}
+            />
+            Show diff
+          </label>
+          <select id="file-select" @change=${this.handleFileChange}>
+            <option value="" ?selected=${!hasCurrentFile} disabled>
+              Files changed:
```

Comment:
```suggestion
              Files changed
```

IMO, it looks nicer without the colon.